### ISSUE TITLE
Need tests around monitoring of capibilities

### ISSUE THREAD

**Issue by HenryMiller1** [icecc/icecream#283]
Issues #282 exposes our lack of testing around the monitor protocol (ie icemon) 

A test was proposed there, but it needs to be  separate issue, we should be able to change our tests:

Create a fake icecream monitor that tracks messages.
After every test completes check the monitor to verify that the count JobLocalBeginMsg matches the count of JobLocalEndMsg. In tests/test.sh look for uses of flush_logs/check_logs_for_generic_errors (there are several other check_logs for errors functions) for places to inject this check.

There may be other checks that we can apply. there is also potential to use the fake monitor to verify that test builds are going to remote machines or not without reading log messages, but this can be considered latter.

**Comment by llunak** [comment]
I don't think this needs a separate fake monitor. The code for sending the messages can log it with trace() and test.sh can verify the correctness.
